### PR TITLE
Barbican

### PR DESCRIPTION
**Pull request by LarsErikP**
This PR brings in barbican!

### Upgrade notes

Barbican is implemented as opt-in. It is not enabled before the hiera-keys with its keystone password is set in hiera

### New hiera-keys

Global keys:
- `ntnuopenstack::barbican::keystone::password` - Barbican's password in keystone
- `ntnuopenstack::barbican::mysql::password` - Barbican's database password
- `ntnuopenstack::barbican::simple_crypto::kek` - Key encryption key. Used in the simple crypto plugin to encrpyt and decrypt secrets. 32 chars, base64-encoded

Node-specific keys:
This must be set in every barbican server's node file
- `apache::mod::wsgi::package_name: 'libapache2-mod-wsgi-py3'`
- `apache::mod::wsgi::mod_path: '/usr/lib/apache2/modules/mod_wsgi.so'`

### CODE REVIEW

File: manifests/keystone/endpoint.pp [hunk 1]
```diff
@@ -32,6 +37,11 @@
     include ::ntnuopenstack::swift::endpoint
   }
 
+  # If there is a password for barbican in hiera, define an endpoint for swift.
```
Comment (obrestad) [inline manifests/keystone/endpoint.pp:40]: Denne kommentaren skaper mye mening :)

File: manifests/barbican.pp [hunk 2]
```diff
@@ -0,0 +1,7 @@
+# Installs and configures the barbican api and worker
+class ntnuopenstack::barbican {
+  include ::ntnuopenstack::barbican::api
+  include ::ntnuopenstack::barbican::crypto
+  include ::ntnuopenstack::barbican::worker
```
Comment (obrestad) [inline manifests/barbican.pp:5]: Burde man her bare inkludert ::barbican::worker i stedet? Sånn siden ::ntnuopenstack::barbican::worker ikke gjør noe annet uansett?

File: manifests/keystone/endpoint.pp (hunk 1 above)
Comment (LarsErikP) [reply manifests/keystone/endpoint.pp:40]: Haha! Nei, den der skapte ikke så mye mening. Var vel litt kopipasta ute og gikk da serru. Men hadde du to kommentarer sa du? Jeg ser bare en?

File: manifests/barbican.pp (hunk 2 above)
Comment (LarsErikP) [reply manifests/barbican.pp:5]: Ja det kan vi jo gjøre. Jeg laget overbygget kun i tilfelle man sku trenge noe mer senere. Jeg vet ikke jeg?

**Pull request merged**